pages/blog/asyncapi-say-hi-to-triggermesh.md

```diff
@@ -0,0 +1,272 @@
+
+```sh
+tmctl create broker TriggerMeshAsyncAPI
+tmctl create source kafka --name orders-kafkasource --topic orders --bootstrapServers host.docker.internal:9092 --groupID orders-group
+tmctl create source httppoller --name ordersorderjson-httppollersource --method GET --endpoint http://host.docker.internal:8000/order.json --interval 10s --eventType io.triggermesh.httppoller.event
+tmctl create source googlecloudpubsub --name orders-gcp-pubsubsource --topic projects/jmcx-asyncapi/topics/ordersgcp --serviceAccountKey $(cat serviceaccountkey.json)
+```
+
+The first command creates a lightweight event broker, the central component that will decouple event producers and consumers, and provide pub/sub style reliable delivery of events to their targets. The name of the broker is derived from the title of the AsyncAPI definition.
+
+Next, one source component is created per `channel` that provides a `subscribe` operation and has a reference to a `server` with a support protocol such as `http`, `kafka`, or `googlepubsub`.
+
+To get this working on your environment, you may want to play with the `bootstrapServers` value for the Kafka source, and the endpoint `host` for the HTTP poller source. You can change them in the AsyncAPI definition and then re-run the parser.
+You'll also need to create a file called `serviceaccountkey.json` with a GCP [service account key](https://cloud.google.com/iam/docs/creating-managing-service-account-keys), if you want to get the Google Pub/Sub channel working. If not, you can delete the Pub/Sub channel in the asyncAPI definition.
+
+You can visit the TriggerMesh documentation for the [Kafka source](https://docs.triggermesh.io/sources/kafka/), [HTTP Poller source](https://docs.triggermesh.io/sources/httppoller/), and [Google Pub/Sub source](https://docs.triggermesh.io/sources/googlecloudpubsub/) for more information on parameters and usage.
+
+We can now execute these `tmctl` commands to create the TriggerMesh components. You can either copy/paste them into your terminal, or run the generated script e.g.:
+
+```sh
+sh tmctl.sh
+```
+Or you can pipe the output of the parser into the shell e.g:
+
+```sh
+node parser.js | sh
+```
+
+We'll also open a second terminal and start `tmctl watch` to watch events flowing through the broker:
+
+```sh
+tmctl watch
+```
+
+## Orders are already coming in from the HTTP service
+
+The first thing you'll notice is that the HTTP poller is starting to produce events from the order management system that exposes the HTTP API, we can see the event showing up in the broker in the `tmctl watch` terminal:
+
+<Figure
+  src="/img/posts/asyncapi-say-hi-to-triggermesh/poller-event.webp"
+  caption="A CloudEvent that originates from the HTTP poller"
+/>
+
+The poller is configured to fetch an event every 10 seconds. You can adjust the endpoint and other parameters depending on your environment and needs. I’m using host.docker.internal because I’m running on Docker Desktop.
```

Comment:
```suggestion
The poller is configured to fetch an event every 10 seconds. You can adjust the endpoint and other parameters depending on your environment and needs. I’m using `host.docker.internal` because I’m running on Docker Desktop.
```